File: 101_Symmetric_Tree/Solution.py
class Solution(object):
    def isSymmetric(self, root):
        if not root:
            return True
        stack = [(root.left, root.right)]
        while stack:
            left, right = stack.pop()
            if not left and not right:
                continue
            if not left or not right or left.val != right.val:
                return False
            stack.append((left.left, right.right))
            stack.append((left.right, right.left))
            
        return True

File: 101_Symmetric_Tree/Solution.java
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null){
            return true;
        }
        return ajudaNoEspelho(root.left, root.right);
    }

    public boolean ajudaNoEspelho(TreeNode left, TreeNode right) {
        if (left == null && right == null) {
            return true;
        }
        if (left == null || right == null || left.val != right.val) {
            return false;
        }
        return ajudaNoEspelho(left.left, right.right) && ajudaNoEspelho(left.right, right.left);
    }

}